src/ctorch.cpp

```diff
@@ -23,36 +23,52 @@ constexpr auto get_dtype(torch_data_t dtype)
   case torch_kFloat64:
     return torch::kFloat64;
   default:
-    std::cerr << "[ERROR]: unknown data type, setting to torch_kFloat32"
+    std::cerr << "[WARNING]: unknown data type, setting to torch_kFloat32"
               << std::endl;
     return torch::kFloat32;
   }
 }
 
-constexpr auto get_device(torch_device_t device)
+const auto get_device(torch_device_t device_type, int device_index)
 {
-  switch (device) {
+  switch (device_type) {
   case torch_kCPU:
-    return torch::kCPU;
+    if (device_index != -1) {
+      std::cerr << "[WARNING]: device index unused for CPU-only runs"
+                << std::endl;
+    }
+    return torch::Device(torch::kCPU);
   case torch_kCUDA:
-    return torch::kCUDA;
+    if (device_index == -1) {
+      std::cerr << "[WARNING]: device index unset, setting to zero"
```

Comment:
```suggestion
      std::cerr << "[WARNING]: device index unset, defaulting to 0"
```
suggestion?
Feel free to ignore.